main.nf

```diff
@@ -1192,40 +1313,43 @@ process IVAR_VARIANTS {
     !params.skip_variants && 'ivar' in callers
 
     input:
-    tuple val(sample), val(single_end), path(bam) from ch_sort_bam_ivar_variants
+    tuple val(sample), val(single_end), path(mpileup) from ch_mpileup_ivar_variants
     path header from ch_ivar_variants_header_mqc
     path fasta from ch_fasta
     path gff from ch_gff
 
     output:
-    tuple val(sample), val(single_end), path("*.vcf.gz*") into ch_ivar_variants_vcf
-    path "*.bcftools_stats.txt" into ch_ivar_variants_bcftools_mqc
-    path "*.counts_mqc.tsv" into ch_ivar_variants_count_mqc
+    tuple val(sample), val(single_end), path("${prefix}.vcf.gz*") into ch_ivar_highfreq_snpeff
+    tuple val(sample), val(single_end), path("${sample}.vcf.gz*") into ch_ivar_lowfreq_snpeff
+    path "${prefix}.bcftools_stats.txt" into ch_ivar_bcftools_highfreq_mqc
+    path "${prefix}.variant.counts_mqc.tsv" into ch_ivar_count_highfreq_mqc
+    path "${sample}.variant.counts_mqc.tsv" into ch_ivar_count_lowfreq_mqc
+    path "${sample}.bcftools_stats.txt"
     path "${sample}.tsv"
     path "*.log"
 
     script:
     features = params.gff ? "-g $gff" : ""
+    prefix = "${sample}.AF${params.max_allele_freq}"
     """
-    samtools mpileup \\
-        -A \\
-        -d 6000000 \\
-        -B \\
-        -Q 0 \\
-        ${bam[0]} \\
-        | ivar variants -p $sample -r $fasta $features
+    cat $mpileup | ivar variants -q $params.min_base_qual -t 0.03 -m 5 -r $fasta -p $sample $features
 
     ivar_variants_to_vcf.py ${sample}.tsv ${sample}.vcf > ${sample}.variant.counts.log
     bgzip -c ${sample}.vcf > ${sample}.vcf.gz
     tabix -p vcf -f ${sample}.vcf.gz
     bcftools stats ${sample}.vcf.gz > ${sample}.bcftools_stats.txt
-
     cat $header ${sample}.variant.counts.log > ${sample}.variant.counts_mqc.tsv
+
+    ivar_variants_to_vcf.py ${sample}.tsv ${prefix}.vcf --pass_only --min_allele_freq $params.max_allele_freq > ${prefix}.variant.counts.log
```

Comment:
Is it better to filter within this process or at a later step?  If we filter within, those filtered out variant calls are not available, even as intermediate results.

Reply:
They should be? I am exporting the filtered calls into a channel here:
https://github.com/drpatelh/nf-core-viralrecon/blob/b99c663deccb5c57f3c41ba6c08f8bc5d9294632/main.nf#L1322